torch/csrc/api/include/torch/nn/modules/rnn.h

```diff
@@ -39,7 +45,7 @@ class RNNImplBase : public torch::nn::Cloneable<Derived> {
       int64_t number_of_gates = 1,
       bool has_cell_state = false);
 
-  std::vector<Variable> forward(std::vector<Variable>);
+  RNNOutput forward(Variable input, Variable state = {});
```

Comment:
Does the AnyModule interact with optional arguments well?

Reply:
No it does not unfortunately, you always have to pass defaults